File: 2348-number-of-zero-filled-subarrays/2348-number-of-zero-filled-subarrays.cpp
class Solution {
public:
    long long zeroFilledSubarray(vector<int>& nums) {
        long long cnt = 0;
        long long streak = 0;

        for(int i=0;i<nums.size();i++){
            if(nums[i]==0){
                streak+=1;
                cnt+=streak;
            }else{
                streak = 0;
            }
        }

        return cnt;
    }
};

File: 2348-number-of-zero-filled-subarrays/2348-number-of-zero-filled-subarrays.py
class Solution(object):
    def zeroFilledSubarray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        cnt = 0
        streak = 0

        for i in nums:
            if i==0:
                streak +=1
                cnt += streak
            else:
                streak = 0
            
        return cnt


        